docs/generated/checks.md

```diff
@@ -1,26 +1,304 @@
- | non-existent-service-account | Yes | Alert on pods referencing a service account that isn't found | Make sure to create the service account, or to refer to an existing service account. | non-existent-service-account | `{}` |
- | privileged-container | Yes | Alert on deployments with containers running in privileged mode | Don't run your container as privileged unless required. | privileged | `{}` |
- | required-annotation-email | No | Alert on objects without an 'email' annotation with a valid email | Add an email annotation to your object with the contact information of the object's owner. | required-annotation | `{"key":"email","value":"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+"}` |
- | required-label-owner | No | Alert on objects without the 'owner' label | Add an email annotation to your object with information about the object's owner. | required-label | `{"key":"owner"}` |
- | run-as-non-root | Yes | Alert on containers not set to runAsNonRoot | Set runAsUser to a non-zero number, and runAsNonRoot to true, in your pod or container securityContext. See https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for more details. | run-as-non-root | `{}` |
- | ssh-port | Yes | Alert on deployments exposing port 22, commonly reserved for SSH access | Ensure that non-SSH services are not using port 22. Ensure that any actual SSH servers have been vetted. | ports | `{"port":22,"protocol":"TCP"}` |
- | unset-cpu-requirements | Yes | Alert on containers without CPU requests and limits set | Set your container's CPU requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details. | cpu-requirements | `{"lowerBoundMillis":0,"requirementsType":"any","upperBoundMillis":0}` |
- | unset-memory-requirements | Yes | Alert on containers without memory requests and limits set | Set your container's memory requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details. | memory-requirements | `{"lowerBoundMB":0,"requirementsType":"any","upperBoundMB":0}` |
- | writable-host-mount | No | Alert on containers that mount a host path as writable | If you need to access files on the host, mount them as readOnly. | writable-host-mount | `{}` |
+## dangling-service
+
+**Enabled by default**: Yes
+
+**Description**: Alert on services that don't have any matching deployments
+
+**Remediation**: Make sure your service's selector correctly matches the labels on one of your deployments.
+
+**Template**: [dangling-service](generated/templates.md#dangling-service)
```

Comment:
@viswajithiii The links to template page doesn't work as expected where template `HumanName` is different from the template name. Can we fix this somehow so that id's match?

Reply:
Hmm, it looks like GitHub doesn't offer you control of the anchor link for headings -- it's going to be derived from the name of the heading, which in our case is the human name.

We can solve this programmatically. I'll comment on the code section.